cmd/acr/purge.go

```diff
@@ -272,19 +279,34 @@ func getTagsToDelete(ctx context.Context,
 			}
 			lastUpdateTime, err = time.Parse(time.RFC3339Nano, *tag.LastUpdateTime)
 			if err != nil {
-				return nil, "", err
+				return nil, "", skippedTagsCount, err
 			}
 			// If a tag did match the regex filter, is older than the specified duration and can be deleted then it is returned
 			// as a tag to delete.
 			if lastUpdateTime.Before(timeToCompare) && *(*tag.ChangeableAttributes).DeleteEnabled {
-				tagsToDelete = append(tagsToDelete, tag)
+				tagsEligibleForDeletion = append(tagsEligibleForDeletion, tag)
+			}
+		}
+
+		tagsToDelete := []acr.TagAttributesBase{}
+
+		if keep == 0 {
```

Comment:
Wondering if `skippedCount` already reached `keep` here, can we avoid checking check one by one?
```go
newLastTag = getLastTagFromResponse(resultTags)

if keep == 0 || skippedTagsCount == keep {
	tagsToDelete = tagsEligibleForDeletion
	return &tagsToDelete, newLastTag, skippedTagsCount, nil
}
```